src/components/trace-element.vue

```diff
@@ -1,17 +1,20 @@
 <template>
   <div class="pexpr" :class="classObj" id="id">
-    <div v-if="labeled" class="self">
+    <div v-if="labeled" class="self" :class="selfClassObj">
       <trace-label :traceNode="traceNode" :minWidth="minWidth"
                    @hover="onHover" @unhover="onUnhover" @click="onClick"
                    @showContextMenu="onShowContextMenu" />
+      <semantics-result v-if="hasResult" :traceNode="traceNode"
+                        @styleUpdate="updateSelfStyle"/>
     </div>
     <div v-if="!isLeaf" ref="children"
          class="children" :class="{vbox: vbox}"
          :hidden="collapsed">
       <trace-element v-for="child in childrenToRender"
                      :id="child.id" :traceNode="child.traceNode" :context="child.context"
                      :currentLR="child.currentLR" :measureInputText="measureInputText"
-                     :isInVBox="child.isInVBox" :eventHandlers="eventHandlers">
+                     :isInVBox="child.isInVBox" :eventHandlers="eventHandlers"
+                     :showResult="hasResult">
```

Comment:
Re: naming -- thanks for the clarification, and I think the new naming makes things much more clear.

The guard I meant was about showing the semantics editors as a whole. But I need to think a bit more about how to do this, so I think what I might do is merge this PR into a separate branch for now, and then once I decide how I want to handle enabling/disabling semantics editing, I can do that and then merge the branch into master.